File: datasets/Part 1 - Data Preprocessing/data_preprocessing_template.R

# Plantilla para el Pre Procesado de Datos

setwd("~/repos/machinelearning-az/datasets/Part 1 - Data Preprocessing")

# Importar el dataset
dataset = read.csv('data/Data.csv')
#dataset = dataset[, 2:3]

# Dividir los datos en conjunto de entrenamiento y conjunto de test
library(caTools) # Instalar con install.packages('caTools')
set.seed(123) # Para que salgan los mismos resultados que en el curso
split = sample.split(dataset$Purchased, SplitRatio = 0.8)
training_set = subset(dataset, split == TRUE)
testing_set = subset(dataset, split == FALSE)

# Escalado de valores
# Hay que indicar las columnas donde hacer el escalado
# No se hacen sobre los valores dummy de las columnas Country y Purchased porque
# no puede operar con factores que convirtio previamente a tipo string
#training_set[, 2:3] = scale(training_set[, 2:3])
#testing_set[, 2:3] = scale(testing_set[, 2:3])

File: datasets/Part 1 - Data Preprocessing/data_preprocessing_template.py
# -*- coding: utf-8 -*-
"""
Created on Tue Apr 26 21:32:38 2022

@author: mudar
"""

# Plantilla de Pre Procesado

# Como importar las librerias
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importar el dataset
dataset = pd.read_csv('data/Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 3].values

# Dividir el dataset en conjunto de entrenamiento y conjunto de testing
# 80% conjunto entrenamiento y 20% conjunto de testing
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=0)

# Escalado de variables
# Para que no haya tanta diferencia en los rangos de
# valores en los datos de las distintas columnas
"""
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)
"""
